libdnf/base/base.cpp

```diff
@@ -48,6 +48,16 @@ Base::Base()
     load_defaults();
 }
 
+Base::Base(std::vector<std::unique_ptr<Logger>> && loggers)
+    : log_router(std::move(loggers)),
+      repo_sack(get_weak_ptr()),
+      rpm_package_sack(get_weak_ptr()),
+      transaction_history(get_weak_ptr()),
+      vars(get_weak_ptr()),
+      p_impl(new Impl(get_weak_ptr())) {
+    load_defaults();
+}
+
```

Comment:
I was thinking if using the constructors delegation wouldn't be better (also on other places in the code, not only this commit). Like having fully defined the constructor with largest number of parameters and other ones delegate to this one, f.e. in this case we could have just something like this:

`Base::Base()
    : Base(std::vector<std::unique_ptr<Logger>>()) {}`

So we avoid the redundancy in the code. What do you think about that?

Reply:
Also we could use just a single constructor with a default value in this case:

`Base(std::vector<std::unique_ptr<Logger>> && loggers = {});`

Reply:
Removing the parameterless constructor will change the public ABI.  But we can still do it.  I will delete it.